### PR TITLE
Adds basic pegout test

### PR DESCRIPTION
**Pull request by jeremy-then**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,11 +13,11 @@
   "license": "GPL3",
   "dependencies": {
     "@rsksmart/bridge-state-data-parser": "^1.2.0",
-    "@rsksmart/bridge-transaction-parser": "^1.1.1",
+    "@rsksmart/bridge-transaction-parser": "git+https://git@github.com/rsksmart/bridge-transaction-parser#c43e9de6004bb2bfbc29549de50fea2e9c1a9a2d",
```
Comment (marcos-iov) [inline package.json:16]: New version is already published in npm

File: package.json [hunk 2]
```diff
@@ -13,11 +13,11 @@
...
     "@rsksmart/btc-eth-unit-converter": "^1.0.0",
     "@rsksmart/btc-rsk-derivation": "^0.0.2",
     "@rsksmart/pmt-builder": "^3.0.0",
-    "@rsksmart/powpeg-redeemscript-parser": "^1.0.0",
+    "@rsksmart/powpeg-redeemscript-parser": "git+https://git@github.com/rsksmart/powpeg-redeemscript-parser#b3306c64b97c4ac85e5c3296f0ad071b3e1b0bed",
```
Comment (marcos-iov) [inline package.json:20]: same

File: package.json [hunk 3]
```diff
@@ -34,14 +34,7 @@
...
     "peglib": "git+https://github.com/rsksmart/rsk-peglib#v1.4.15",
-    "precompiled-arrowhead600": "npm:@rsksmart/rsk-precompiled-abis@^6.0.0-ARROWHEAD",
-    "precompiled-fingerroot500": "npm:@rsksmart/rsk-precompiled-abis@5.0.0-FINGERROOT",
-    "precompiled-hop400": "npm:@rsksmart/rsk-precompiled-abis@5.0.0-HOP",
-    "precompiled-iris300": "npm:@rsksmart/rsk-precompiled-abis@^4.0.0-IRIS",
-    "precompiled-lovell700": "git+https://github.com/rsksmart/precompiled-abis#master",
-    "precompiled-orchid": "npm:@rsksmart/rsk-precompiled-abis@^1.0.0-ORCHID",
-    "precompiled-papyrus200": "npm:@rsksmart/rsk-precompiled-abis@^3.0.0-PAPYRUS",
-    "precompiled-wasabi100": "npm:@rsksmart/rsk-precompiled-abis@^2.0.0-WASABI",
+    "@rsksmart/rsk-precompiled-abis": "npm:@rsksmart/rsk-precompiled-abis@^6.0.0-ARROWHEAD",
```
Comment (marcos-iov) [inline package.json:37]:
```suggestion
    "@rsksmart/rsk-precompiled-abis": "^6.0.0-ARROWHEAD",
```
isn't that enough?

File: lib/rsk-utils.js [hunk 4]
```diff
@@ -410,6 +412,59 @@ const findEventInBlock = async (rskTxHelper, eventName, fromBlockHashOrBlockNumb
   return null;
 };
 
+/**
+ * Gets all the bridge events in the block range from `fromBlockHashOrBlockNumber` to `toBlockHashOrBlockNumber`.
+ * If no `fromBlockHashOrBlockNumber` is provided, it will start from the latest block number up to the `toBlockHashOrBlockNumber`.
+ * If no `toBlockHashOrBlockNumber` is provided, it will be equal to `toBlockHashOrBlockNumber`.
```
Comment (marcos-iov) [inline lib/rsk-utils.js:418]: something's wrong here

File: lib/rsk-utils.js [hunk 5]
```diff
@@ -410,6 +412,59 @@ const findEventInBlock = async (rskTxHelper, eventName, fromBlockHashOrBlockNumb
   return null;
 };
 
+/**
+ * Gets all the bridge events in the block range from `fromBlockHashOrBlockNumber` to `toBlockHashOrBlockNumber`.
+ * If no `fromBlockHashOrBlockNumber` is provided, it will start from the latest block number up to the `toBlockHashOrBlockNumber`.
```
Comment (marcos-iov) [inline lib/rsk-utils.js:417]: if no `from` value is provided it starts from the latest? Where can it go from there?

File: lib/tests/2wp.js [hunk 6]
```diff
@@ -679,6 +682,55 @@ const execute = (description, getRskHost) => {
...
+
+      // Create a pegin for the serder to ensure there is enough funds to pegout and because this is the natural process
+      const senderRecipientInfo = await createSenderRecipientInfo(rskTxHelper, btcTxHelper);
+      const peginValueInSatoshis = btcToSatoshis(0.5);
+      const btcPeginTxHash = await sendPegin(rskTxHelper, btcTxHelper, senderRecipientInfo.btcSenderAddressInfo, satoshisToBtc(peginValueInSatoshis));
+      await ensurePeginIsRegistered(rskTxHelper, btcPeginTxHash);
+
+      const initial2wpBalances = await get2wpBalances(rskTxHelper, btcTxHelper);
+      const initialSenderAddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo.btcSenderAddressInfo.address);
+      const pegoutValueInRbtc = 0.005;
```
Comment (marcos-iov) [inline lib/tests/2wp.js:697]: I believe that is not the minimum value

File: lib/tests/2wp.js [hunk 7]
```diff
@@ -679,6 +682,55 @@ const execute = (description, getRskHost) => {
...
+      // Callback to get the bridge state after the pegout is created
+      const pegoutCreatedCallback = async () => {
+        bridgeStateAfterPegoutCreation = await getBridgeState(rskTxHelper.getClient());
+      };
+
+      const callbacks = {
+        pegoutCreatedCallback
+      };
+
+      await triggerRelease(rskTxHelpers, btcTxHelper, callbacks);
```
Comment (marcos-iov) [inline lib/tests/2wp.js:716]: For another story, but maybe we could get rid of the callbacks passed to `triggerRelease`? I think it adds confusion and are probably not needed now that we got rid of regression testing

**Review by marcos-iov — COMMENTED** [review]
Nice test! Very complete. Well done

File: lib/rsk-utils.js (hunk 5 above)
Comment (jeremy-then) [reply lib/rsk-utils.js:417]: Updated.

File: lib/tests/2wp.js (hunk 6 above)
Comment (jeremy-then) [reply lib/tests/2wp.js:697]: Updated.

File: lib/tests/2wp.js (hunk 7 above)
Comment (jeremy-then) [reply lib/tests/2wp.js:716]: In this case it is necessary if we want to get the state of the Bridge right when the pegout is waiting for confirmations.

File: package.json (hunk 1 above)
Comment (jeremy-then) [reply package.json]: Updated.

File: package.json (hunk 2 above)
Comment (jeremy-then) [reply package.json]: Updated.

File: package.json (hunk 3 above)
Comment (jeremy-then) [reply package.json]: Updated.

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**